[#226] Add PR template for releases (#227)
serokell/tezos-packaging: .github/PULL_REQUEST_TEMPLATE/release_pull_request_template.md

## Description

This PR introduces a new <!-- insert new version here --> release in tezos-packaging.

<!-- Depending on the type of the release some of the points below can be ommited.

E.g. in case we introduce a new changes to our native packages, it's not necessary to bump
used Tezos sources, create a new release in this repository and update brew formulas.

> Feel free to cross out points that aren't needed in some specific case.

Reviewer: I think we shouldn't do this, if need to make e.g. a second PR after the opam packages are updated it would be great to be able to look at the previous one and see which points where missing IMO

Reviewer: In fact, I think here we should add something like `Resolves `, but instead since it might relate to a previous PR we should have `Follows #`

Note that opam-repository is usually updated with some delay, so CI may fail
due to lack of updated opam Tezos packages. In such case feel free to comment
out native-packaging-related CI steps from the pipeline until the opam-repository
is updated.
-->

> ### Things to do in the release PR

Reviewer: ```suggestion
### Changes related to the creation of a release (conditional)
```

> - [ ] Update Tezos sources and opam-repository (if needed) revisions in [sources.json](../../tree/master/nix/nix/sources.json).

Reviewer: A bit of a nitpick, but usually template checklist are written as if the author replies to question, so for example this should be:
```suggestion
- [ ] I updated Tezos sources and opam-repository (if needed) revisions in [sources.json](../../tree/master/nix/nix/sources.json).
```
can you do the same for the others?

- [ ] Remove old bottles hashes from the brew formulas in [Formula directory](../../tree/master/Formula).

> - [ ] Update versions in brew formulas.

Reviewer: ```suggestion
- [ ] Update `url :tag` and `version`s in brew formulas.
```

- [ ] Update release number in [meta.json](../../tree/master/meta.json).

#### In case the new Tezos release provides a new protocol and corresponding testnet

- [ ] Support new protocol in [protocols.json](../../tree/master/protocols.json).
- [ ] Support new protocol and testnet in native packaging.
- [ ] Support new protocol and testnet in brew formulas.
- [ ] Add tests for the new protocol and testnet.

> ### Things to do after the release PR is merged

Reviewer: ```suggestion
### Things to do after this release PR is merged
```

> 

Reviewer: I suggest adding a comment here specifying that these can be checked in this same PR after the fact.

But maybe that's clear already, so I'll let you choose.

#### Create a new release in this repository

- [ ] Create a new release that is based on the latest autorelease created by the CI.

#### Update brew bottles and repository mirrors

- [ ] Compile brew bottles using [`build-bottles.sh`](../../tree/scripts/build-bottles.sh) script and upload them to the created release.

> - [ ] Add new bottles sha256 hashes to the brew formulas.

Reviewer: Can't these technically be done in the same PR, before it gets merged (when possible)?

If so I think they should be moved above

Author: > Can't these technically be done in the same PR, before it gets merged (when possible)?

It can, but usually we do this only once the bottles are uploaded to avoid potential confusion for users that will try to download a new bottle and get 404.

- [ ] Push changes to either [tezos-packaging-rc](https://github.com/serokell/tezos-packaging-rc) or
      [tezos-packaging-stable](https://github.com/serokell/tezos-packaging-stable) mirror repositories

#### Publish new native packages

Once [opam-repository](https://opam.ocaml.org/packages/) is updated with the new version of Tezos packages.

> - [ ] Publish new Ubuntu packages, see [this instructions](../../tree/master/docker#source-packages-and-publishing-them-on-launchpad-ppa).

Reviewer: ```suggestion
- [ ] Publish new Ubuntu packages, see [these instructions](../../tree/master/docker#source-packages-and-publishing-them-on-launchpad-ppa).
```

> - [ ] Publish new Fedora packages, see [this instructions](../../tree/master/docker#srcrpm-packages).

Reviewer: ```suggestion
- [ ] Publish new Fedora packages, see [these instructions](../../tree/master/docker#srcrpm-packages).
```

## Description

This PR introduces a new <!-- insert new version here --> release in tezos-packaging.

<!-- Depending on the type of the release some of the points below can be ommited.

E.g. in case we introduce a new changes to our native packages, it's not necessary to bump
used Tezos sources, create a new release in this repository and update brew formulas.

Note that opam-repository is usually updated with some delay, so CI may fail
due to lack of updated opam Tezos packages. In such case feel free to comment
out native-packaging-related CI steps from the pipeline until the opam-repository
is updated.
-->

<!-- List issues which are going to be resolved in the new packages from this
release. Write 'None' if there are no related issues.

For example:

> Follows #202, #210

Reviewer 2: It says to list issues that are going to be resolved by the (eventual) release, but I don't think "Follows" represents resolving well enough :thinking: Since AFAIU it _is_ possible to make a release all in one PR, I'd say it should probably still default to "Resolves", and this comment should explain that if something is missing e.g. because of the opam, the line should be changed to "Follows #".

Author: The problem is that "Resolves" will close the mentioned issues once the PR is merged. However, some of the issues should be considered resolved only once updated native packages are published

-->

Follows #

### Changes related to the creation of a release (conditional)

- [ ] I updated Tezos sources and opam-repository (if needed) revisions in [sources.json](../../tree/master/nix/nix/sources.json).
- [ ] I removed old bottles hashes from the brew formulas in [Formula directory](../../tree/master/Formula).
- [ ] I updated `url :tag` and `version`s in brew formulas..
- [ ] I updated release number in [meta.json](../../tree/master/meta.json).

#### In case the new Tezos release provides a new protocol and corresponding testnet

- [ ] I supported new protocol in [protocols.json](../../tree/master/protocols.json).
- [ ] I supported new protocol and testnet in native packaging.
- [ ] I supported new protocol and testnet in brew formulas.
- [ ] I added tests for the new protocol and testnet.

### Things to do after this release PR is merged

<!--

> Some of the changes are done after the new release is created, consider checking
> the merged release PR with this template and follow unfinished points.
> -->

Reviewer 2: ```suggestion
Some of the changes are done after the new release is created, consider following 
and checking unfinished points in the merged release PR using this template.
-->
```
It took me a minute to parse otherwise.

#### Create a new release in this repository

- [ ] I created a new release that is based on the latest autorelease created by the CI.

#### Update brew bottles and repository mirrors

> - [ ] I compiled brew bottles using [`build-bottles.sh`](../../tree/scripts/build-bottles.sh) script and upload them to the created release.

Reviewer 2: ```suggestion
- [ ] I compiled brew bottles using [`build-bottles.sh`](../../tree/scripts/build-bottles.sh) script and uploaded them to the created release.
```
